packages/mds-db/sql-utils.ts

```diff
@@ -46,6 +65,55 @@ export class MDSPostgresClient extends PostgresClient {
   public setConnected(connected: boolean) {
     this.connected = connected
   }
+
+  // Runs a query and returns the selected rows as an array of objects of type R.
+  private async select<R extends QueryResultRow>(command: string, values: (string | number)[]) {
+    await logSql(command, values)
+    const result: QueryResult<R> = await this.query(command, values)
+    return result.rows
+  }
+
+  public async selectAll<R extends QueryResultRow>(command: string, values: (string | number)[] = []) {
+    return this.select<R>(command, values)
+  }
+
+  // Runs a query and returns a single selected row as an object of type R. An exception
+  // is thrown if there are no selected rows or more than one selected row.
+  public async selectOne<R extends QueryResultRow>(command: string, values: (string | number)[] = []) {
```

Comment:
Just curious, I thought you were mostly against throwing an exception if something was not found? Having to remember whether to use `selectOne` or `selectOneOrNull` seems like unnecessary mental overhead.

Reply:
I'm mostly against not having the choice. There's not much overhead in having the ability to ask for one where you expect there to be exactly one and only one vs when there may be one or none. Anyone familiar with Entity Framework will recognize the semantics.